backend/src/main/java/com/javahelp/backend/query/IUserQueryConstraint.java

```diff
@@ -0,0 +1,25 @@
+package com.javahelp.backend.query;
+
+import com.javahelp.model.survey.SurveyResponse;
+import com.javahelp.model.user.User;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public interface IUserQueryConstraint {
+    /**
+     *
+     * @param constraints {@link List} of {@link Constraint}s to be applied for query.
+     * @return {@link Set} of Users that satisfy the list of {@link Constraint}s.
+     */
+    Map<String, User> getProvidersWithConstraints(List<Constraint> constraints);
+
+    /**
+     *
+     * @param constraints{ @link List} of {@link Constraint}s to be applied for query.
+     * @return {@link Map} of {@link String} ID of users and their corresponding
+     * {@link SurveyResponse}s.
+     */
+    Map<String, SurveyResponse> getResponsesWithConstraints(List<Constraint> constraints);
+}
```

Comment:
I don't see a ton of use in this interface, since it's implementation is in terms of other interfaces and abstract classes. Adding another layer of abstraction might not be needed here.